TheSadRogue.Primitives/Rectangle.cs

```diff
@@ -903,5 +904,130 @@ public IEnumerable<Point> MaxXPositions()
+        /// </summary>
+        /// <returns>
+        /// An IEnumerable of either Top and Bottom halves, or Left and Right halves,
+        /// at indexes 0 and 1 respectively.
+        /// </returns>
+        /// <remarks>
+        /// Cuts the rectangle by reducing it's longest dimension by half. For example, a
+        /// rectangle that extends from (3, 3) to (18, 7) that calls DivideInHalf will
+        /// return an IEnumerable with two rectangles. The rectangle at index 0
+        /// starts at (3, 3) and extends to (9, 7), and the rectangle at index 1 extends from
+        /// (10, 3) to (18, 7)
+        /// </remarks>
+        public IEnumerable<Rectangle> Bisect()
+        {
+            if (Width > Height)
+                return BisectVertically();
+
+            else if (Width < Height)
+                return BisectHorizontally();
+
+            else
+                return BisectHorizontally();
+        }
+
+        /// <summary>
+        /// Bisects the rectangle into top and bottom halves
+        /// </summary>
+        /// <returns>An IEnumerable with Top and Bottom Rectangles in indexes 0 and 1, respectively</returns>
+        public IEnumerable<Rectangle> BisectHorizontally()
+        {
+            int startX = MinExtentX;
+            int stopY = MaxExtentY;
+            int startY = MinExtentY;
+            int stopX = MaxExtentX;
+            int bisection = (startY + stopY) / 2;
+
+            yield return new Rectangle(new Point(startX, startY), new Point(stopX, bisection));
+            yield return new Rectangle(new Point(startX, bisection + 1), new Point(stopX, stopY));
+        }
+
+        /// <summary>
+        /// Divides the rectangle into a left and right half.
+        /// </summary>
+        /// <returns>
+        /// An IEnumerable with the Left and Right rectangles in index 0 and 1 respectively.
+        /// </returns>
+        public IEnumerable<Rectangle> BisectVertically()
+        {
+            int startY = MinExtentY;
+            int stopY = MaxExtentY;
+            int startX = MinExtentX;
+            int stopX = MaxExtentX;
+            int bisection = (startX + stopX) / 2;
+
+            yield return new Rectangle(new Point(startX, startY), new Point(bisection, stopY));
+            yield return new Rectangle(new Point(bisection + 1, startY), new Point(stopX, stopY));
+        }
+
+        /// <summary>
+        /// Divides one rectangle by another, and returns an IEnumerable of rectangles the size of divisor that fit
+        /// within the dividend rectangle.
+        /// </summary>
+        /// <param name="dividend">The rectangle being divided</param>
+        /// <param name="divisor">The rectangle by which to divide</param>
+        /// <returns>
+        /// IEnumerable of Rectangles the size of divisor, that fit within the dividend.
+        /// </returns>
+        public static IEnumerable<Rectangle> operator /(Rectangle dividend, Rectangle divisor)
```

Comment:
I think having an operator that returns a iterator in C# might create some issues.  See comment in unit tests.